persistence/src/main/java/cz/muni/fi/dao/DriverDaoImpl.java

```diff
@@ -65,24 +65,27 @@ public Driver findTestDriver(String name, String surname) {
         }
     }
 
-    public void addDriver(Driver driver) {
+    public Driver addDriver(Driver driver) {
         if(driver == null){
             throw new IllegalArgumentException("argument is null");
         }
         em.persist(driver);
```

Comment:
pls improve this. add checks for null parameters of Driver. also note that id must be null here